cirq-web/cirq_ts/src/circuit/grid_circuit.ts

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 The Cirq Developers
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Group} from 'three';
+import {GridQubit} from './components/grid_qubit';
+import {TwoQubitGate, SingleQubitGate, GridCoord} from './components/types';
+
+/**
+ * Class that gathers serialized circuit information
+ * from a Python cirq.Circuit and reconstructs it to be
+ * displayed using three.js
+ */
+export class GridCircuit extends Group {
+  readonly moments: number;
```

Comment:
Can we allow the moments to vary with the length since the client dynamically adds gates to the circuit?

Reply:
A still outstanding issue. I wonder about this number as well - why do we need it at all? It seems like that this is just used as the initial default moment number for the qubits... Consider just passing it in to the private `addQubit` method and then we don't even have to store it. Also rename to `initial_num_moments`.

Reply:
I thought it would be useful for error checking like in `addSymbol()`, where I say if the user tries to add a Symbol to a moment outside of the bounds then throw an error. This way, we don't have to keep passing it in to different functions. However, that check is only really useful for dev purposes, so I can remove it.